.github/workflows/build-binaries.yml

```diff
@@ -15,7 +25,92 @@ jobs:
             xcode: "15.2.0"
     steps:
       - uses: actions/checkout@v4
+
       - name: Switch to Xcode ${{ matrix.xcode }}
         uses: maxim-lobanov/setup-xcode@v1.6.0
         with:
           xcode-version: ${{ matrix.xcode }}
+
+      - name: Get Core Version
+        id: get-core-version
+        run: |
+          REALM_CORE_VERSION=${{ inputs.core-version }}
+          if [[ -z $REALM_CORE_VERSION ]]; then
+            REALM_CORE_VERSION=$(sed -n 's/^REALM_CORE_VERSION=\(.*\)$/\1/p' dependencies.list)
+          fi
+          echo "version=$REALM_CORE_VERSION" >> "$GITHUB_OUTPUT"
+
+      - name: Checkout Core
+        uses: actions/checkout@v4
+        with:
+          repository: realm/realm-core
+          path: core
+          submodules: recursive
```

Comment:
We shouldn't need core's submodules

Reply:
Those are needed to build the prebuilds. Otherwise, I get errors like this one when running `build-apple-device.sh`:

```
CMake Error at external/CMakeLists.txt:4 (add_subdirectory):
  The source directory

    /Users/runner/work/realm-swift/realm-swift/core/external/catch

  does not contain a CMakeLists.txt file.
```